### DISCUSSION TITLE
Is it possible to connect data in a notebook server on kubernetes with a kubeflow pipeline?

### DISCUSSION THREAD

**Discussion by nutmilk10** [kubeflow/pipelines#10688] [General]
I have kubeflow running on an on-prem cluster where I have a jupyter notebook server with a data volumne '/data' that has a file called sample.csv. I want to be able to read the csv in my kubeflow pipeline. Here is what my kubeflow pipeline looks like, not sure how I would integrate my csv from my notebook server. Any help would be appreciated.

```
from kfp import components


def read_data(csv_path: str):
    import pandas as pd
    df = pd.read_csv(csv_path)
    return df

def compute_average(data: list) -> float:
    return sum(data) / len(data)

# Compile the component
read_data_op = components.func_to_container_op(
                                func=read_data,
                                output_component_file='read_data_component.yaml',
                                base_image='python:3.7',  # You can specify the base image here
                                packages_to_install=["pandas"])

compute_average_op = components.func_to_container_op(func=compute_average,
                                output_component_file='compute_average_component.yaml',
                                base_image='python:3.7',
                                packages_to_install=[])
```

**Comment by rimolive** [comment]
You can do this with any remote path, not only s3. Take a look at the [Kubeflow quickstart](https://www.kubeflow.org/docs/components/pipelines/v2/installation/quickstart/) documentation, the dataset is downloaded from a website, then it will be managed by the internal artifact tracking subcomponent.
